lib/logstash/outputs/logstash.rb

```diff
@@ -1,58 +1,51 @@
 # encoding: utf-8
 
-require 'logstash/outputs/base'
-require 'logstash/namespace'
+require "logstash/outputs/base"
+require "logstash/namespace"
 
-require "logstash/plugin_mixins/plugin_factory_support"
+require "logstash/plugin_mixins/http_client"
 require "logstash/plugin_mixins/validator_support/required_host_optional_port_validation_adapter"
+require "zlib"
 
 class LogStash::Outputs::Logstash < LogStash::Outputs::Base
   extend LogStash::PluginMixins::ValidatorSupport::RequiredHostOptionalPortValidationAdapter
 
-  include LogStash::PluginMixins::PluginFactorySupport
+  include LogStash::PluginMixins::HttpClient[:with_deprecated => false]
+
+  require "logstash/utils/load_balancer"
 
   config_name "logstash"
 
   # Sets the host of the downstream Logstash instance.
   # Host can be any of IPv4, IPv6 (requires to be in enclosed bracket) or host name, the forms:
-  #     `"127.0.0.1"`
-  #     `"127.0.0.1:9800"`
-  #     `"foo-bar.com"`
-  #     `"foo-bar.com:9800"`
-  #     `"[::1]"`
-  #     `"[::1]:9000"`
+  #     `"127.0.0.1"` or `["127.0.0.1"]` if single host with default port
+  #     `"127.0.0.1:9801"` or `["127.0.0.1:9801"]` if single host with custom port
+  #     `["foo-bar.com", "foo-bar.com:9800"]`
+  #     `["[::1]", "[::1]:9000"]`
   #     `"[2001:0db8:85a3:0000:0000:8a2e:0370:7334]"`
   #
-  # NOTE: `hosts` naming is intentional and multi-host support is planned.
-  #
-  config :hosts, :validate => :required_host_optional_port, :required => true
-
-  # optional username/password credentials
-  config :username, :validate => :string,   :required => false
-  config :password, :validate => :password, :required => false
-
-  config :ssl_enabled,                 :validate => :boolean, :default => true
+  config :hosts, :validate => :required_host_optional_port, :list => true, :required => true
 
-  # SSL:IDENTITY:SOURCE cert/key pair
-  config :ssl_certificate,             :validate => :path
-  config :ssl_key,                     :validate => :path
+  config :username, :validate => :string, :required => false
```

Comment:
review note: mixin-http plugin has `user`/`password` configs. I personally prefer `username` instead `user`. I don't know if it is intentionally (is it related with proxy?!) named `user`. If so let's change this, otherwise current behaviour accepts both `user` and `username` but overwrites `user` with `username`.

@edmocosta do you have any information related to this case?